apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.ExecutableCommand.cs

```diff
@@ -445,23 +447,26 @@ private ReadRequest GetReadRequest()
 
             private ReadOrQueryRequest GetReadOrQueryRequest()
             {
-                if (Partition != null)
+                ReadOrQueryRequest request = Partition switch
                 {
-                    return Partition.Request;
-                }
+                    null => CommandTextBuilder.SpannerCommandType switch
+                    {
+                        SpannerCommandType.Select => ReadOrQueryRequest.FromQueryRequest(GetExecuteSqlRequest()),
+                        SpannerCommandType.Read => ReadOrQueryRequest.FromReadRequest(GetReadRequest()),
+                        _ => throw new InvalidOperationException($"Implementation error: Invalid command type ${CommandTextBuilder.SpannerCommandType} for read or query. This should not happen.")
+                    },
+                    _ => DirectedReadOptions switch
+                    {
+                        null => Partition.Request,
```

Comment:
I think this merits a comment. I believe I understand the purpose, but it's a bit subtle.

(Can we use ValueTuple in our library? If so, I wonder whether switching on `(Partition, DirectedReadOptions)` would be worth doing?)

Reply:
Definitely happy to go with this for now. I might have a go at tweaking it later, but the comments make things much clearer already.